openeo/metadata.py

```diff
@@ -522,3 +523,61 @@ def _repr_html_(self):
     def __str__(self) -> str:
         bands = self.band_names if self.has_band_dimension() else "no bands dimension"
         return f"CollectionMetadata({self.extent} - {bands} - {self.dimension_names()})"
+
+
+def metadata_from_stac(url: str) -> CubeMetadata:
+    """
+    Reads the band metadata a static STAC catalog or a STAC API Collection and returns it as a :py:class:`CubeMetadata`
+
+    :param url: The URL to a static STAC catalog (STAC Item, STAC Collection, or STAC Catalog) or a specific STAC API Collection
+    :return: A :py:class:`CubeMetadata` containing the DataCube band metadata from the url.
+    """
+
+    def get_band_names(asst: pystac.Asset) -> List[Band]:
+        return [Band(eo_band["name"]) for eo_band in asst.extra_fields["eo:bands"]]
```

Comment:
`Band` also has `common_name` and `wavelength_um` fields, so I would also extract these if they are available